packages/babel-plugin-transform-typescript/src/enum.ts

```diff
@@ -156,7 +156,10 @@ export function translateEnumValues(
       } else {
         const initializerPath = memberPath.get("initializer");
 
-        if (initializerPath.isReferencedIdentifier()) {
+        if (
+          initializerPath.isIdentifier() &&
+          initializerPath.isReferencedIdentifier()
+        ) {
           ReferencedIdentifier(initializerPath, {
```

Comment:
`isReferencedIdentifier` ensures that `this` is either an `Identifier` or a `JSXIdentifier`, but `ReferencedIdentifier` can't handle `JSXIdentifier`.

Reply:
I wonder why our `NodePath` has the wrong variance and `NodePath<A | B>` is not inferred as the same as `NodePath<A> | NodePath<B>` :thinking: